### PR TITLE
Implement parsing of arguments for create event feature

### PR DESCRIPTION
**Pull request by jordankanghm**
Also added some storage files to their respective packages.

Lastly, updated the user guide for create events feature.

### CODE REVIEW

File: src/main/java/seedu/address/model/event/DateTime.java [hunk 1]
```diff
@@ -66,36 +67,37 @@ public static boolean isValidDateTime(String dateAndTime) {
...
+        int hour = Integer.parseInt(time.substring(0, 2));
+        int min = Integer.parseInt(time.substring(2, 4));
+
+        return LocalDateTime.of(year, month, day, hour, min);
     }
 
     @Override
     public String toString() {
-        return dateTime.toString();
+        return dateAndTime.toString();
```
Comment (brein62) [inline src/main/java/seedu/address/model/event/DateTime.java:100]: see my PR -- the parsing method for DateTime is changed

**Review by brein62 — CHANGES_REQUESTED** [review]
overall refactoring LGTM, but i won't merge yet due to conflicts in the implementation of DateTime

File: src/main/java/seedu/address/model/event/DateTime.java (hunk 1 above)
Comment (brein62) [reply src/main/java/seedu/address/model/event/DateTime.java:100]: we try to reconcile these changes before we merge anything

**Comment by jordankanghm** [comment]
lgtm

**Review by brein62 — APPROVED** [review]
looks good, really good lookin

**Comment by jordankanghm** [comment]
THANK YOU

**Comment by brein62** [comment]
looks good, the most good lookin so far

File: src/main/java/seedu/address/model/event/Description.java [hunk 2]
```diff
@@ -14,13 +14,14 @@ public class Description {
     /*
      * The first character of the description must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
+     * The expression allows for multiple words to be inputted.
      */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final String VALIDATION_REGEX = "[A-Za-z0-9 _.,!\"'/$]*";
```
Comment (ruo-x) [inline src/main/java/seedu/address/model/event/Description.java:19]: wowie look so complex

File: src/main/java/seedu/address/model/event/EventName.java [hunk 3]
```diff
@@ -4,44 +4,45 @@
...
+ * Guarantees: immutable; is valid as declared in {@link #isValidEventName(String)}
  */
 public class EventName {
 
     public static final String MESSAGE_CONSTRAINTS =
             "Names should only contain alphanumeric characters and spaces, and it should not be blank";
 
     /*
-     * The first character of the address must not be a whitespace,
+     * The first character of the name must not be a whitespace,
```
Comment (ruo-x) [inline src/main/java/seedu/address/model/event/EventName.java:16]: can consider changing name to event name?

**Review by ruo-x — APPROVED** [review]
damn tiny suggestion to further improve the clarity of javadoc.

File: src/main/java/seedu/address/model/event/EventName.java (hunk 3 above)
Comment (jordankanghm) [reply src/main/java/seedu/address/model/event/EventName.java:16]: ok i change for u

**Pull request merged**